src/main/java/com/google/sps/classes/Utility.java

```diff
@@ -29,4 +34,50 @@ public static String convertToJsonUsingGson(Object object) {
     Gson gson = new Gson();
     return gson.toJson(object);
   }
+
+  /*
+   * Class that returns the ID of a logged in user.
+   */
+  public static int getUserId() {
+    int userId = -1;
+
+    // Get logged in user email.
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+
+    // If user is not logged in, return -1.
+    if (email = "") {
+      return userId;
+    }
+
+    // Set up variables needed to connect to MySQL database.
+    String url = "jdbc:mysql://localhost:3306/Mintern?useSSL=false&serverTimezone=PST8PDT";
+    String user = "root";
+    String password = "";
+
+    // Set up query to check if user is already registered.
+    String query = "SELECT id FROM User WHERE email = ?";
+
+    try {
+      // Establish connection to MySQL database.
+      Connection connection = DriverManager.getConnection(url, user, password);
+
+      // Create the MySQL prepared statement, execute it, and store the result.
+      PreparedStatement preparedStatement = connection.prepareStatement(query);
+      preparedStatement.setString(1, email);
```

Comment:
What is the 1 for? Can you leave a comment explaining what that represents? We typically avoid hardcoded numbers, so if one does show up you'll want to leave a comment explaining why/what it is.

Reply:
Is this comment good enough: `Create the MySQL prepared statement, add email variable to the query, execute it, and store the result.`? The `query` string variable has a certain number of question marks (in this case only 1), so what the setString does is replace the question mark at the position of the first parameter (again, in this case 1). I didn't know if I had to explain all that in the comment, so I just wrote that, but let me know if it still needs a change!

Reply:
Ah I see, okay how about this as a comment: "Takes the query specified above and sets the email field to the logged in user's email."

Otherwise I would have had no idea that this code was replacing the ? in the query!

Reply:
Done!